google/cloud/firestore_v1/query.py

```diff
@@ -386,6 +391,61 @@ def stream(
                 last_snapshot = snapshot
                 yield snapshot
 
+    def stream(
+        self,
+        client=None,
+        transaction=None,
+        limit=None,
+        retry: retries.Retry = gapic_v1.method.DEFAULT,
+        timeout: float = None,
+    ) -> Type["firestore_v1.query.Iterator"]:
+        """Read the documents in the collection that match this query.
+
+        This sends a ``RunQuery`` RPC and then returns an iterator which
+        consumes each document returned in the stream of ``RunQueryResponse``
+        messages.
+
+        .. note::
+
+           The underlying stream of responses will time out after
+           the ``max_rpc_timeout_millis`` value set in the GAPIC
+           client configuration for the ``RunQuery`` API.  Snapshots
+           not consumed from the iterator before that point will be lost.
+
+        If a ``transaction`` is used and it already has write operations
+        added, this method cannot be used (i.e. read-after-write is not
+        allowed).
+
+        Args:
+            client
+                (Optional[:class:`~google.cloud.firestore_v1.client.Client`]):
+                The client used to make this query. If unset, use the client
+                that created this query.
+            transaction
+                (Optional[:class:`~google.cloud.firestore_v1.transaction.Transaction`]):
+                An existing transaction that this query will run in.
+            limit (Optional[int]):
+                The maximum number of documents the query is allowed to return.
+            retry (google.api_core.retry.Retry): Designation of what errors, if any,
+                should be retried.  Defaults to a system-specified policy.
+            timeout (float): The timeout for this request.  Defaults to a
+                system-specified value.
+        """
+    
+        if client is None:
+            client = self._client
+
+        return Iterator(
+            self,
+            self._parent,
+            client,
+            transaction=transaction,
+            limit=limit,
+            retry=retry,
+            timeout=timeout,
+            all_descendants=self._all_descendants,
+        )
```

Comment:
You could implement this as:

```
def stream(*args, **kwargs):
      inner_generator = self._old_stream(*args, **kwargs)
      return Iterator(inner_generator) 
```

And then the new custom Iterator would just wrap the old one:

```
class Iterator:

    def __init__(response_generator):
        self._generator = response_generator

    def __iter__():
        return self._response_generator
```

Reply:
This is a much simpler way to make an iterator! I updated the code to use this method.